File: 3738-make-array-non-decreasing/3738-make-array-non-decreasing.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumPossibleSize = function(nums) {
    let n = nums.length, cnt = 1;
    let num = nums[0];
    for (let i = 1; i < n; ++i) {
        if (nums[i] >= num) {
            num = nums[i];
            cnt++;
        }
    }
    return cnt;
};

File: 3738-make-array-non-decreasing/3738-make-array-non-decreasing.go
func maximumPossibleSize(nums []int) int {
    n, cnt, num := len(nums), 1, nums[0]
    for i := 1; i < n; i++ {
        if nums[i] >= num {
            cnt++
            num = nums[i]
        }
    }
    return cnt
}